services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java

```diff
@@ -1139,10 +1145,35 @@ public void ignore() {
                 }
             };
 
+            evaluateApplicationInitiatedActionTrigger(session, provider, model, authSession);
             provider.evaluateTriggers(result);
         }
     }
+    
+    // Determine if provider is being requested as an Application-Initiated Action
+    // If so, add it to the authSession.
+    private static void evaluateApplicationInitiatedActionTrigger(final KeycloakSession session,
+                                                                  final RequiredActionProvider provider, 
+                                                                  final RequiredActionProviderModel model,
+                                                                  final AuthenticationSessionModel authSession
+                                                                  ) {
+        if (provider.initiatedActionSupport() == InitiatedActionSupport.NOT_SUPPORTED) return;
+        
+        String aia = authSession.getClientNote(AIA_REQUEST);
+        if (aia == null) return;
 
+        // make sure you are evaluating the action that was requested
+        if (!aia.equalsIgnoreCase(model.getProviderId())) return;
+        
+        if (session.getContext().getClient().getRole(AccountRoles.INITIATE_ACTION) == null) {
```

Comment:
I would remove AccountRoles.INITIATE_ACTION and remove this check. At this point no authentication of the client has happened so this is pretty much meaningless as someone can just use a client_id from a different app (such as Keycloak account console).

Checking the client and that it has permissions (manage_account I'd say rather than a separate role) is only needed for the follow-up work where a consent screen is required, but can be bypassed by clients that is permitted to do so (and can supply a valid id_token_hint to "authenticate" itself).

Reply:
Code now checks manage_account role.  Not sure if you meant checking manage_account would only be for phase II stuff, but it seems to make sense for phase I as well.